src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerChangeServiceTest.java

```diff
@@ -269,21 +291,65 @@ void testCollateAllBeneficialOwnerChanges() {
     assertTrue(result.stream().anyMatch(OtherBeneficialOwnerChange.class::isInstance));
   }
 
+  @Test
+  void testCollateAllBeneficialOwnerChangesNoChipsReference() {
+    // setup corporate DTO
+    BeneficialOwnerCorporateDto beneficialOwnerCorporateDto = new BeneficialOwnerCorporateDto();
+    beneficialOwnerCorporateDto.setName("John Smith");
+
+    // setup individual DTO
+    BeneficialOwnerIndividualDto beneficialOwnerIndividualDto = new BeneficialOwnerIndividualDto();
+    beneficialOwnerIndividualDto.setFirstName("John");
+    beneficialOwnerIndividualDto.setLastName("Doe");
+
+    // setup other DTO
+    BeneficialOwnerGovernmentOrPublicAuthorityDto beneficialOwnerOtherDto = new BeneficialOwnerGovernmentOrPublicAuthorityDto();
+    beneficialOwnerOtherDto.setName("John Doe");
+
+    when(publicPrivateBo.get(beneficialOwnerCorporateDto.getChipsReference())).thenReturn(
+        mockPublicPrivateBoPair);
+    when(publicPrivateBo.get(beneficialOwnerIndividualDto.getChipsReference())).thenReturn(
+        mockPublicPrivateBoPair);
+    when(publicPrivateBo.get(beneficialOwnerOtherDto.getChipsReference())).thenReturn(
+        mockPublicPrivateBoPair);
+
+    when(overseasEntitySubmissionDto.getBeneficialOwnersCorporate()).thenReturn(
+        List.of(beneficialOwnerCorporateDto));
+    when(overseasEntitySubmissionDto.getBeneficialOwnersIndividual()).thenReturn(
+        List.of(beneficialOwnerIndividualDto));
+    when(overseasEntitySubmissionDto.getBeneficialOwnersGovernmentOrPublicAuthority()).thenReturn(
+        List.of(beneficialOwnerOtherDto));
+
+    PrintStream standardOut = System.out;
+    ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(outputStreamCaptor));
+    
+    List<Change> result = beneficialOwnerChangeService.collateBeneficialOwnerChanges(
+        publicPrivateBo, overseasEntitySubmissionDto);
+
+    System.setOut(standardOut);
+
+    assertEquals("", outputStreamCaptor.toString());
```

Comment:
What's the purpose of this? We just want to assert the collateBeneficialOwnerChanges method returns an empty list of changes when the BOs have no chips reference

Reply:
The purpose was to make sure that it wasn't doing any log messages 

With this one, we want the logic that gets rid of the BOs to happen in the stream and not further down.

If there for whatever reason is a Chips Reference it doesn't recognise it will return null but also there will be a log message 

By checking there are no log messages we can check that this is happening where it is supposed to be.